File: BusinessLogic/src/main/java/name/benjaminabbitt/wwwwwh/url/ServerBuilderListener.java
package name.benjaminabbitt.wwwwwh.url;

import io.grpc.ServerBuilder;
import io.grpc.services.HealthStatusManager;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;

import javax.inject.Singleton;

@Singleton
public class ServerBuilderListener implements BeanCreatedEventListener<ServerBuilder<?>> {
    @Override
    public ServerBuilder<?> onCreated(BeanCreatedEvent<ServerBuilder<?>> event) {
        final ServerBuilder<?> builder = event.getBean();
        HealthStatusManager healthStatusManager = new HealthStatusManager();
        builder.addService(healthStatusManager.getHealthService());
        builder.maxInboundMessageSize(1024);
        //TODO: set port
        return builder;
    }
}

File: BusinessLogic/src/main/java/name/benjaminabbitt/wwwwwh/url/ServerBuilderListener.kt
package name.benjaminabbitt.wwwwwh.url

import io.grpc.ServerBuilder
import io.grpc.services.HealthStatusManager
import io.micronaut.context.event.BeanCreatedEvent
import io.micronaut.context.event.BeanCreatedEventListener
import javax.inject.Singleton

@Singleton
class ServerBuilderListener : BeanCreatedEventListener<ServerBuilder<*>> {
    override fun onCreated(event: BeanCreatedEvent<ServerBuilder<*>>): ServerBuilder<*> {
        val builder = event.bean
        val healthStatusManager = HealthStatusManager()
        builder.addService(healthStatusManager.healthService)
        // Unclear why this is not required.  Introspection?  Including this line ends up with double instantiation of BookmarkService
        ///builder.addService(ctx.getBean(BookmarkService.class));
        return builder
    }
}